mantidproject/mantidimaging
# Extends loading to read log files and 180 degree projection

Redesigns the loading dialog UI and refactors it to have view and presenter. The functionality is expanded to search for:
- 180 degree projection
- IMAT Log files for sample and flat images

180 degree projections are now loaded as a new stack. If a 180 degree projection is loaded, then any filter applied to the sample will also be applied to the 180 degree projection stack.

The current format of the log files is being parsed, and only the projection angles read. It is likely the log file format will change (hopefully to CSV), so the parsing will have to be adapted.

To Test:
- Selecting sample should still work and find flat/dark files. Additionally 180 degree projection and sample/flat log files should be found
  - Check that all the options on the loading dialog work, e.g. that unchecking `Use` does in fact ignore the entry, indices are properly used
- Load some data containing only sample and flat/dark images, do some filtering and reconstruct, everything should run normally
- Load some data that also contains a 180 degree projection. Apply some filters to the **sample** data and check that they are correctly applied to the **180 degree** projection also.
- Open the reconstruction window and run Auto find using `Correlation`
 
Fixes #496 
Fixes #486
Fixes #485 
Fixes #478
